Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp

```diff
@@ -381,25 +398,73 @@ namespace AzToolsFramework
         return QPoint{ denormalizedX, denormalizedY };
     }
 
+    void wrapCursorX(const QRect& rect, QPoint& point) {
+        if (rect.left() < point.x()) 
+        {
+            point.setX(rect.right() - 1);
+        }
+        else if (rect.right() > point.x()) 
+        {
+            point.setX(rect.left() + 1);
+        }
+    }
+
+    void wrapCursorY(const QRect& rect, QPoint& point) {
+        if (rect.top() < point.y()) 
+        {
+            point.setY(rect.bottom() - 1);
+        }
+        else if (rect.bottom() > point.y()) 
+        {
+            point.setY(rect.top() + 1);
+        }
+    }
```

Comment:
update: Please run `clang-format` over this file (braces always go on a new line) and also functions always begin with a capital letter

Also these functions are perfect for quick unit tests to ensure they're doing the right thing